caluma/data_source/data_sources.py

```diff
@@ -37,11 +43,23 @@ class BaseDataSource:
     """
 
     info = None
-    timeout = 3600
-    default = []
+    default = None
 
     def __init__(self):
         pass
 
     def get_data(self, info):  # pragma: no cover
         raise NotImplementedError()
+
+    def __call__(self, info):
```

Comment:
I'm not a fan of unnecessary use of magic functions. Why not name it `try_get_data_with_fallback()` or similar?

Reply:
I share your concern regarding magic functions. The reason I went with that is, that we don't want to reserve method-names. I guess if we go with `try_get_data_with_fallback()` the risk is very small.

Reply:
In JS, I'd prefix the method with an underscore. Does a similar convention exist in python?

Reply:
Yes, but then it's considered a "private"-method and should not be part of the API.